dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -160,6 +160,7 @@ bool TarArchive::EndOfArchive() const {
 
 void TarArchive::SeekArchive(int64_t offset) {
   assert(offset % T_BLOCKSIZE == 0);
+  eof_ = false;
```

Comment:
Maybe `ParseHeader` should set that. Now it is a bit inconsistent between `SetEof` which hides handling of `eof_ = true` and here.

Reply:
eof_ prevents the ParseHeader from killing itself by accidentally reading from a stream that has reached the end